std/installer/test.ts

```diff
@@ -1,472 +0,0 @@
-  "%~dp0\deno.exe" "run" "--allow-net" "--allow-read" "http://localhost:4500/installer/testdata/echo.ts" "--foobar" %*
-) ELSE (
-  @SETLOCAL
-  @SET PATHEXT=%PATHEXT:;.TS;=;%
-  "deno" "run" "--allow-net" "--allow-read" "http://localhost:4500/installer/testdata/echo.ts" "--foobar" %*
-)
-`
-      /* eslint-enable max-len */
-    );
-  }
-
-  assertEquals(
-    await fs.readFileStr(filePath),
-    /* eslint-disable max-len */
-    `#!/bin/sh
-# This executable is generated by Deno. Please don't modify it unless you know what it means.
-basedir=$(dirname "$(echo "$0" | sed -e 's,\\\\,/,g')")
-
-case \`uname\` in
-  *CYGWIN*) basedir=\`cygpath -w "$basedir"\`;;
-esac
-
-if [ -x "$basedir/deno" ]; then
-  "$basedir/deno" "run" "--allow-net" "--allow-read" "http://localhost:4500/installer/testdata/echo.ts" "--foobar" "$@"
-  ret=$?
-else
-  "deno" "run" "--allow-net" "--allow-read" "http://localhost:4500/installer/testdata/echo.ts" "--foobar" "$@"
-  ret=$?
-fi
-exit $ret
-`
-    /* eslint-enable max-len */
-  );
-});
-
-installerTest(async function installLocalModuleAndRun(): Promise<void> {
-  const tempDir = await makeTempDir();
-  const localModule = path.join(Deno.cwd(), "installer", "testdata", "echo.ts");
-  await install("echo_test", localModule, ["hello"], tempDir);
-
-  const filePath = path.resolve(tempDir, "echo_test");
-  const fileInfo = await stat(filePath);
-  assert(fileInfo.isFile());
-
-  const ps = run({
-    args: [filePath + (path.isWindows ? ".cmd" : ""), "foo"],
-    stdout: "piped"
-  });
-
-  if (!ps.stdout) {
-    assert(!!ps.stdout, "There should have stdout.");
-    return;
-  }
-
-  let thrown = false;
-
-  try {
-    const b = await readAll(ps.stdout);
-
-    const s = new TextDecoder("utf-8").decode(b);
-
-    assertEquals(s, "hello, foo");
-  } catch (err) {
-    console.error(err);
-    thrown = true;
-  } finally {
-    await remove(tempDir, { recursive: true });
-    ps.close();
-  }
-
-  assert(!thrown, "It should not throw an error");
-}, true); // set true to install module in your real $HOME dir.
-
-installerTest(async function installAndMakesureItCanRun(): Promise<void> {
-  const tempDir = await makeTempDir();
-  await install(
-    "echo_test",
-    "http://localhost:4500/installer/testdata/echo.ts",
-    ["hello"],
-    tempDir
-  );
-
-  const filePath = path.resolve(tempDir, "echo_test");
-  const fileInfo = await stat(filePath);
-  assert(fileInfo.isFile());
-
-  const ps = run({
-    args: [filePath + (path.isWindows ? ".cmd" : ""), "foo"],
-    stdout: "piped"
-  });
-
-  if (!ps.stdout) {
-    assert(!!ps.stdout, "There should have stdout.");
-    return;
-  }
-
-  let thrown = false;
-
-  try {
-    const b = await readAll(ps.stdout);
-
-    const s = new TextDecoder("utf-8").decode(b);
-
-    assertEquals(s, "hello, foo");
-  } catch (err) {
-    console.error(err);
-    thrown = true;
-  } finally {
-    await remove(tempDir, { recursive: true });
-    ps.close();
-  }
```

Comment:
@ry not sure how to test it, those tests depend on `deno` being in `PATH`...

Reply:
Can you add target/debug to the PATH? (`current_executable().parent().unwrap()`)

Reply:
That's doable, but tests are run in parallel, I'm not sure messing with env variables is a good idea :/

Reply:
So the test is to actually call the shell script? You'll have to do this with a custom integration test (non-itest). When you spawn the subprocess to execute the shell script, you'll be able to specify the env.

Reply:
Obiviously... thanks 👍 still not sure what causes panic in current tests